api/krusty/configmaps_test.go

```diff
@@ -182,22 +182,20 @@ data:
   BIRD: falcon
   MOUNTAIN: everest
   OCEAN: pacific
-  forces.txt: |2
-
```

Comment:
I think this is related to the other bug (https://github.com/kubernetes-sigs/kustomize/pull/4004#discussion_r653803898). Something weird is happening with line breaks - hopefully I can narrow down the problem.

Reply:
I'm not sure why the string even had two new lines to begin with anyways. It's generated from a file called forces.txt with the contents

```
gravitational
electromagnetic
strong nuclear
weak nuclear
```

So I'm not sure if it makes sense for an extra newline to be added. Maybe this is a bug fix? I get the same result even when I don't disable the sequence indent.

Reply:
Doesn't the file start with a newline? It's being created from a string a few lines up here, and it looks like it starts with a newline:

```go
th.WriteF("forces.txt", `
gravitational
electromagnetic
strong nuclear
weak nuclear
`)
```

Reply:
Hmm I think you're right. Because this is a problem with go-yaml.v3 at head we'll need to do some digging to find the root cause here. I can do that tomorrow, but this might be something we have to swallow as part of the bump if we can't get it fixed in time.

Reply:
Fixed in my fork!